### PR TITLE
add columnExpression

### PR DESCRIPTION
**Pull request by nevermore3**
background  is  https://www.yuque.com/nebulagraph/emchnb/kg2u5m

### CODE REVIEW

File: src/common/context/ExpressionContext.h [hunk 1]
```diff
@@ -61,6 +61,9 @@ class ExpressionContext {
     // Get Edge
     virtual Value getEdge() const = 0;
 
+    // Get Value by Column number
```
Comment (yixinglu) [inline src/common/context/ExpressionContext.h:64]: column index

File: src/common/expression/ColumnExpression.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    const Value& eval(ExpressionContext &ctx) override;
+
+    void accept(ExprVisitor *visitor) override;
+
+    std::unique_ptr<Expression> clone() const override {
+        return std::make_unique<ColumnExpression>(index_);
+    }
+
+    std::string toString() const override {
+        return "COLUMN";
```
Comment (yixinglu) [inline src/common/expression/ColumnExpression.h:31]: `COLUMN[index_]`

File: src/common/expression/ColumnExpression.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    std::string toString() const override {
+        std::stringstream out;
+        out << "COLUMN[" << index_ << "]";
+        return out.str();
+    }
+
+    bool operator==(const Expression &expr) const override {
+        return kind() == expr.kind();
```
Comment (darionyaphet) [inline src/common/expression/ColumnExpression.h:37]: only compare type? why don't compare the index value?

File: src/common/expression/ColumnExpression.cpp [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const Value& ColumnExpression::eval(ExpressionContext &ctx) {
+    result_ = ctx.getColumn(index_);
+    return result_;
+}
+
+bool ColumnExpression::operator==(const Expression &expr) const {
+    if (kind_ != expr.kind()) {
+        return false;
+    }
+    const auto &r = dynamic_cast<const ColumnExpression &>(expr);
```
Comment (yixinglu) [inline src/common/expression/ColumnExpression.cpp:22]: static_cast<>()

**Review by CPWstatic — DISMISSED** [review]
LGTM. Please address @yixinglu's comment.

**Review by CPWstatic — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**